api-controllers/ServiceFabrikApiController.js

```diff
@@ -433,33 +369,39 @@ class ServiceFabrikApiController extends FabrikBaseController {
   abortLastBackup(req, res) {
     req.manager.verifyFeatureSupport('backup');
     const backupStartedAt = new Date();
-    return eventmesh.apiServerClient.getLastOperation({
-      resourceId: req.params.instance_id,
-      operationName: CONST.OPERATION_TYPE.BACKUP,
-      operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-    }).then(backupGuid => {
-      return eventmesh.apiServerClient.getOperationState({
+    return eventmesh
+      .apiServerClient.getLastOperation({
+        resourceId: req.params.instance_id,
         operationName: CONST.OPERATION_TYPE.BACKUP,
         operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-        operationId: backupGuid,
-      }).then(state => {
-        // abort only if the state is in progress
-        if (state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS) {
-          return eventmesh.apiServerClient.updateOperationState({
+      })
+      .then(backupGuid => {
+        return eventmesh
+          .apiServerClient.getOperationState({
             operationName: CONST.OPERATION_TYPE.BACKUP,
             operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
             operationId: backupGuid,
-            stateValue: CONST.OPERATION.ABORT
-          });
-        } else {
-          logger.info(`Skipping abort as state is : ${state}`);
-        }
-      }).then(() => ServiceFabrikApiController.getResourceOperationStatus({
-        operationId: backupGuid,
-        start_state: CONST.OPERATION.ABORT,
-        started_at: backupStartedAt
-      }));
-    }).then(status => res.status(status.state === 'aborting' ? CONST.HTTP_STATUS_CODE.ACCEPTED : CONST.HTTP_STATUS_CODE.OK).send({}));
+          })
+          .then(state => {
```

Comment:
why cant the methods apiServerClient.getLastOperation & apiServerClient.getOperationState be merged into one as a generic method like `getLastOperationState`? Just a thought, i think this might be useful at many other places in many other controllers/pollers as well later down the line.

Reply:
Yes, we are having a ApiServerClient refactoring coming up. It has too many wrappers now